src/pip/_internal/commands/uninstall.py

```diff
@@ -64,6 +64,8 @@ def run(self, options, args):
             )
             if req.name:
                 reqs_to_uninstall[canonicalize_name(req.name)] = req
+            else:
+                raise InstallationError("Invalid requirement: '{}'".format(name))
```

Comment:
I don't know if we want to `raise` as a long term goal but we certainly don't want in the next release.
We should first log a warning stating that the invalid requirement is ignored : 
`"Invalid requirement: '{}'".format(name) ignored - the uninstall command expects named requirements.`

And depending on what is decided regarding raising in a future version we might want to add: `This will become an error in pip 21.X.`

Reply:
And rereading #4958 nobody seems to push for an error so starting with a `logger.warning` alone should be fine :)

Reply:
Ok, however in the case of logging the error, an error will be raised later in the form of:
```
ERROR: You must give at least one requirement to uninstall (see "pip help uninstall")
```
So, for example. in the case of an archive as a parameter, `install_req_from_line()` will log a warning, later on we will log a warning and an error will be `raised` causing the following log:
```
WARNING: Requirement 'swh.loader.tar' looks like a filename, but the file does not exist
WARNING: Invalid requirement: 'swh.loader.tar' ignored - the uninstall command expects named requirements.
ERROR: You must give at least one requirement to uninstall (see "pip help uninstall")
```
I do not want to change the function flow to avoid this case but I believe this situation is worth mentioning.
@xavfernandez currently I just implemented the logging part, if we'll decide to keep it - I'll implement the testing and rebase from `master`.

Reply:
Keep in mind that you can pass several packages to `uninstall`:

`pip uninstall pkg-abc http://example.com/other_pkg.zip --yes`
would output:
```
WARNING: Invalid requirement: 'http://example.com/other_pkg.zip' ignored - the uninstall command expects named requirements.
Successfully uninstalled pkg-abc-2.0
```

Reply:
Gotcha, @xavfernandez . I fixed the tests according to my understanding (I check whether the log is as expected and make sure the `raise` clause won't cause the test to fail). I'm waiting for your response 😅 
Btw, can you remind me of the rebase action needed to be done to take care of the failing tests?